integration_tests/ctesque/src/androidTest/java/android/app/ContextTest.java

```diff
@@ -203,4 +204,56 @@ public void alarmManager_instance_retrievesSameAlarmClockInfo() {
           });
     }
   }
+
+  @Test
+  public void cameraManager_applicationInstance_isNotSameAsActivityInstance() {
+    CameraManager applicationCameraManager =
+        (CameraManager)
+            ApplicationProvider.getApplicationContext().getSystemService(Context.CAMERA_SERVICE);
+    try (ActivityScenario<TestActivity> scenario = ActivityScenario.launch(TestActivity.class)) {
+      scenario.onActivity(
+          activity -> {
+            CameraManager activityCameraManager =
+                (CameraManager) activity.getSystemService(Context.CAMERA_SERVICE);
+            assertThat(applicationCameraManager).isNotSameInstanceAs(activityCameraManager);
+          });
+    }
+  }
+
+  @Test
+  public void cameraManager_activityInstance_isSameAsActivityInstance() {
+    try (ActivityScenario<TestActivity> scenario = ActivityScenario.launch(TestActivity.class)) {
+      scenario.onActivity(
+          activity -> {
+            CameraManager activityCameraManager =
+                (CameraManager) activity.getSystemService(Context.CAMERA_SERVICE);
+            CameraManager anotherActivityCameraManager =
+                (CameraManager) activity.getSystemService(Context.CAMERA_SERVICE);
+            assertThat(anotherActivityCameraManager).isSameInstanceAs(activityCameraManager);
+          });
+    }
+  }
+
+  @Test
+  public void cameraManager_instance_retrievesSameCameraIdList() throws Exception {
+    CameraManager applicationCameraManager =
+        (CameraManager)
+            ApplicationProvider.getApplicationContext().getSystemService(Context.CAMERA_SERVICE);
+    try (ActivityScenario<TestActivity> scenario = ActivityScenario.launch(TestActivity.class)) {
+      scenario.onActivity(
+          activity -> {
+            CameraManager activityCameraManager =
+                (CameraManager) activity.getSystemService(Context.CAMERA_SERVICE);
+
+            try {
+              String[] applicationCameraIdList = applicationCameraManager.getCameraIdList();
```

Comment:
> The camera list is empty or not empty?

Its not empty 

07-20 23:13:35.334  6918  6918 I System.out: Application Camera ID List: [0, 1]
07-20 23:13:35.334  6918  6918 I System.out: Activity Camera ID List: [0, 1]

Reply:
> What about adding a new assertion to ensure camera list is not empty?

Hi @utzcoz , is there something I am missing??? because in my local every test is passing but here instrument tests are failing

Reply:
> > SEVERE: Execute android.app.ContextTest.cameraManager_instance_retrievesSameCameraIdList: FAILED
> > expected not to be empty
> 
> Looks like Emulator runs on CI doesn't have camera.

So what needs to be done to merge this PR?

Reply:
> > > SEVERE: Execute android.app.ContextTest.cameraManager_instance_retrievesSameCameraIdList: FAILED
> > > expected not to be empty
> > 
> > 
> > Looks like Emulator runs on CI doesn't have camera.
> 
> So what needs to be done to merge this PR?

Gentle reminder @utzcoz @hoisie

Reply:
> Remove checking for camera id list.

I have removed the entire cameraManager_instance_retrievesSameCameraIdList test since it focused on checking and comparing the camera ID list between different CameraManager instances

please review